resources/js/features/user/settings/components/AvatarSection/AvatarSection.tsx

```diff
@@ -0,0 +1,119 @@
+import type { FC } from 'react';
+import { LuAlertCircle } from 'react-icons/lu';
+
+import { BaseButton } from '@/common/components/+vendor/BaseButton';
+import { baseCardTitleClassNames } from '@/common/components/+vendor/BaseCard';
+import {
+  BaseFormControl,
+  BaseFormField,
+  BaseFormItem,
+  BaseFormLabel,
+  BaseFormMessage,
+  BaseFormProvider,
+} from '@/common/components/+vendor/BaseForm';
+import { BaseInput } from '@/common/components/+vendor/BaseInput';
+import { toastMessage } from '@/common/components/+vendor/BaseToaster';
+
+import { usePageProps } from '../../hooks/usePageProps';
+import { useResetNavbarUserPic } from '../../hooks/useResetNavbarUserPic';
+import { useAvatarSectionForm } from './useAvatarSectionForm';
+import { useResetAvatarMutation } from './useResetAvatarMutation';
+
+export const AvatarSection: FC = () => {
+  const { can } = usePageProps<App.Community.Data.UserSettingsPageProps>();
+
+  const { form, mutation: formMutation, onSubmit } = useAvatarSectionForm();
+
+  const resetAvatarMutation = useResetAvatarMutation();
+
+  const { resetNavbarUserPic } = useResetNavbarUserPic();
+
+  const handleResetAvatarClick = () => {
+    if (
+      !confirm(
+        'Are you sure you want to reset your avatar to the default? This cannot be reversed.',
+      )
+    ) {
+      return;
+    }
+
+    toastMessage.promise(resetAvatarMutation.mutateAsync(), {
+      loading: 'Resetting...',
+      success: () => {
+        resetNavbarUserPic();
+
+        return 'Reset avatar!';
+      },
+      error: 'Something went wrong.',
+    });
+  };
+
+  return (
+    <div className="flex flex-col gap-4">
+      <h3 className={baseCardTitleClassNames}>Avatar</h3>
+
+      {can.updateAvatar ? (
+        <>
+          <p>Only png, jpeg, and gif files are supported.</p>
+
+          <BaseFormProvider {...form}>
+            <form onSubmit={form.handleSubmit(onSubmit)} className="flex flex-col gap-1">
+              <div>
+                <BaseFormField
+                  control={form.control}
+                  name="imageData"
+                  render={({ field }) => (
+                    <BaseFormItem className="flex flex-col gap-2">
+                      <BaseFormLabel className="text-menu-link">New Image</BaseFormLabel>
+
+                      <BaseFormControl>
+                        <BaseInput
```

Comment:
I feel like this should be a button. It's not obvious that you need to click Choose File. I immediately went to the Upload button expecting it to allow me to select a file.
![image](https://github.com/user-attachments/assets/466b07ef-42c8-4899-b5f2-40d67474dfc5)

Reply:
As an HTML native file input, this took some doing to make it look like a button when it's actually not, but in latest it should be a fairly close approximation:

![Screenshot 2024-09-02 at 4 12 19 PM](https://github.com/user-attachments/assets/85fa9673-6bdd-43c6-b4ab-21500d3a0b4e)

Additionally, it's appropriate for the Upload button on this particular form to be disabled if there's no file input, so I've made that change. This should help mitigate people reaching immediately for that button.